packages/angular/cli/upgrade/version.ts

```diff
@@ -61,6 +65,27 @@ export class Version {
       process.exit(2);
     }
 
+    try {
+      const angularCliPkgPath = require.resolve('@angular/cli/package.json', resolveOptions);
+      angularCliPkgJson = require(angularCliPkgPath);
+      if (!(angularCliPkgJson && angularCliPkgJson['version'])) {
+        throw new Error();
+      }
+    } catch (error) {
+      console.error(terminal.bold(terminal.red(tags.stripIndents`
+        Cannot determine versions of "@angular/cli".
+        This likely means your local installation is broken. Please reinstall your packages.
+      `)));
+      process.exit(2);
+    }
+
+    const angularCliVersion = new Version(angularCliPkgJson['version']);
+    const upperBound = angularCliVersion.major;
+    const lowerBound = upperBound - 1;
+    // e.g. CLI 8.0 supports '>=7.0.0 <9.0.0 || ^7.0.0-beta.0 || ^8.0.0-beta.0'.
+    const supportedAngularSemver = `>=${lowerBound}.0.0 <${upperBound + 1}.0.0` +
+      ` || ^${lowerBound}.0.0-beta.0 || ^${upperBound}.0.0-beta.0`;
```

Comment:
The `semver` package supports the `includePrerelease ` option now:  https://github.com/npm/node-semver#functions

Reply:
SGTM, changed.